File: baekjoon/String/5525.cpp
// 백준 5525번 : IOIOI
#include <iostream>

using namespace std;

int main() {
    int N, M, ans = 0;
    string str;

    cin >> N >> M >> str;

    for (int i = 0; i < M; i++) {
        char ch = str[i];
        int cnt = 0;
        if (ch == 'I') {
            while (str[i + 1] == 'O' && str[i + 2] == 'I') {
                cnt++;
                if (cnt == N) {
                    ans++;
                    cnt--;
                }
                i += 2;
            }
        } 
    }
    cout << ans;
    return 0;
}

File: baekjoon/String/5525.py
# 백준 5525번 : IOIOI
if __name__ == "__main__":
    N = int(input())
    M = int(input())
    S = list(input())

    cnt = ans = i = 0

    while i < M - 2:
        if S[i] == 'I' and S[i + 1] == 'O' and S[i + 2] == 'I':
            cnt += 1
        
            if cnt == N:
                cnt -= 1
                ans += 1
            i += 2

        else:
            cnt = 0
            i += 1

    print(ans)